engine/crates/engine-v2/src/operation/logical_planner/selection_set.rs

```diff
@@ -515,24 +562,30 @@ impl<'schema, 'a> SelectionSetLogicalPlanner<'schema, 'a> {
                     .then(probe_field.query_position().cmp(&extra_query_position))
             })
             .expect_err("extra field cannot be present already");
-        field_ids.insert(i, id);
+
+        field_ids.insert(extra_field_id, id);
         self.operation[parent_selection_set_id].field_ids_ordered_by_parent_entity_id_then_position = field_ids;
     }
 
     fn generate_all_candidates<'field>(
         &mut self,
-        unplanned_fields: &HashMap<FieldId, FieldDefinition<'schema>>,
+        unplanned_fields: &mut HashMap<FieldId, FieldDefinition<'schema>>,
         planned_selection_set: &mut PlannedSelectionSet,
         candidates: &mut HashMap<ResolverDefinitionId, ChildPlanCandidate<'schema>>,
     ) -> LogicalPlanningResult<()>
     where
         'schema: 'field,
     {
-        for (&id, definition) in unplanned_fields {
+        let mut plan_changes = Vec::new();
```

Comment:
I would rename it to `interface_fields_to_replan` or something like this, `plan_changes` is very general while it's use is quite specific